packages/db/index.js

```diff
@@ -735,6 +747,139 @@ export class DBClient {
+    /** @type {{data: {cid: string}}} */
+    const { data: content } = await this._client
+      .from('content')
+      .select('cid')
+      .eq('cid', pinRequest.requestedCid)
+      .single()
+
+    const toInsert = {
+      requested_cid: pinRequest.requestedCid,
+      auth_key_id: pinRequest.authKey,
+      name: pinRequest.name
+    }
+
+    // If content already exists updated foreigh key
+    if (content?.cid) {
+      toInsert.content_cid = content.cid
+    }
+
+    /** @type {{data: import('./db-client-types').PAPinRequestItem, error: PostgrestError }} */
+    const { data, error } = await this._client
+      .from(PAPinRequestTableName)
+      .insert(toInsert)
+      .select(pinRequestSelect)
+      .single()
+
+    if (error) {
+      throw new DBError(error)
+    }
+
+    return normalizePaPinRequest(data)
+  }
+
+  /**
+   * Get a Pin Request by id
+   *
+   * @param {number} pinRequestId
+   * @return {Promise<import('./db-client-types').PAPinRequestUpsertOutput>}
+   */
+  async getPAPinRequest (pinRequestId) {
+    /** @type {{data: import('./db-client-types').PAPinRequestItem, error: PostgrestError }} */
+    const { data, error } = await this._client
+      .from(PAPinRequestTableName)
+      .select(pinRequestSelect)
+      .eq('id', pinRequestId)
+      .single()
+
+    if (error) {
+      throw new DBError(error)
+    }
+
+    return normalizePaPinRequest(data)
+  }
+
+  /**
+   * Creates some content and relative pins, pin_sync_request and pin_requests
+   *
+   * Once the content is created through this function, cron jobs will run to check the
+   * pin status and update them in our db.
+   * At the same time a pin_request is created to duplicate the content on Pinata
+   *
+   * @param {import('./db-client-types').ContentInput} content
+   * @param {object} [opt]
+   * @param {boolean} [opt.updatePinRequests] If provided
+   * @return {Promise<string>} The content cid
+   */
+  async createContent (content, {
+    updatePinRequests = false
+  } = {}) {
+    const now = new Date().toISOString()
+
+    /** @type {{data: string, error: PostgrestError }} */
+    const { data: cid, error: createError } = await this._client
+      .rpc('create_content', {
+        data: {
+          content_cid: content.cid,
+          dag_size: content.dagSize,
+          inserted_at: now,
+          updated_at: now,
+          pins: content.pins.map(pin => ({
+            status: pin.status,
+            location: {
+              peer_id: pin.location.peerId,
+              peer_name: pin.location.peerName,
+              region: pin.location.region
+            }
+          }))
+        }
+      }).single()
+
+    if (createError) {
+      throw new DBError(createError)
+    }
+
+    // Update Pin Request FK
+    if (updatePinRequests) {
+      /** @type {{data: string, error: PostgrestError }} */
+      const { error: updateError } = (await this._client
+        .from(PAPinRequestTableName)
```

Comment:
Just a note that in NFT.Storage (and also in [pinata](https://docs.pinata.cloud/api-pinning/pinning-services-api)) it's referred to as PSA. It would be good to be consistent 😁

Reply:
Done in #810